skiko/src/iosMain/kotlin/org/jetbrains/skiko/SkiaLayer.ios.kt

```diff
@@ -121,18 +122,10 @@ actual open class SkiaLayer(
         this.picture = PictureHolder(picture, pictureWidth.toInt(), pictureHeight.toInt())
     }
 
-    fun draw() {
-        contextHandler?.apply {
-            if (!initContext()) {
-                error("initContext() failure")
-            }
-            initCanvas()
-            clearCanvas()
-            val picture = picture
-            if (picture != null) {
-                drawOnCanvas(picture.instance)
-            }
-            flush()
+    internal fun draw(canvas: Canvas) {
+        check(!isDisposed) { "SkiaLayer is disposed" }
+        picture?.also {
+            canvas.drawPicture(it.instance)
```

Comment:
BTW, what benefits we currently have from having intermediate picture instead of drawing to canvas with regular draw calls?

Reply:
Drawing into picture is needed only if we draw into another thread. I keep ios code intact for now, as it out of scope the PR.

Reply:
OK